TeamSnapSDK/SDK/DataLayer/TSDKObjectsRequest.m

```diff
@@ -73,35 +73,106 @@
+                               [TSDKEvent class],
+                               [TSDKContactPhoneNumber class],
+                               [TSDKMemberPayment class],
+                               [TSDKTrackedItem class],
+                               [TSDKTslMetadatum class],
+                               [TSDKSport class],
+                               [TSDKContact class],
+                               [TSDKEventLineupEntry class],
+                               [TSDKMemberPhoto class],
+                               [TSDKSportPosition class],
+                               [TSDKMemberPreferences class],
+                               [TSDKMessage class],
+                               [TSDKInvoicesAggregate class],
+                               [TSDKCustomDatum class],
+                               [TSDKTrackedItemStatus class],
+                               [TSDKForumPost class],
+                               [TSDKAssignment class],
+                               [TSDKEventLineup class],
+                               [TSDKMember class],
+                               [TSDKTeam class],
+                               [TSDKContactEmailAddress class],
+                               [TSDKinvitationFinder class],
+                               [TSDKTeamStatistic class],
+                               [TSDKTeamMediaGroup class],
+                               [TSDKTeamMediumComment class],
+                               [TSDKStatisticDatum class],
+                               [TSDKUser class],
+                               [TSDKTeamMedium class],
+                               [TSDKMemberAssignment class],
+                               [TSDKCustomField class],
+                               [TSDKMessagingPermission class],
+                               [TSDKStatisticGroup class],
+                               [TSDKMemberStatistic class],
+                               [TSDKTslScore class],
+                               [TSDKMemberPhoneNumber class],
+                               [TSDKSmsGateway class],
+                               [TSDKDivisionPreferences class],
+                               [TSDKLeagueCustomDatum class],
+                               [TSDKCountry class],
+                               [TSDKMessageDatum class],
+                               [TSDKBroadcastAlert class],
+                               [TSDKEventStatistic class],
+                               [TSDKTslChat class],
+                               [TSDKTeamPreferences class],
+                               [TSDKLocation class],
+                               [TSDKDivisionLocation class]];
+    });
+    return _supportedSDKObjects;
+}
+
++ (NSArray *)dynamicSupportedObjectsList {
+    NSMutableArray *supportedObjects = [[NSMutableArray alloc] init];
+
+    unsigned int classCount = 0;
+    Class *classList = objc_copyClassList(&classCount);
+
+    for (unsigned int i = 0; i < classCount; i++) {
```

Comment:
Let's upgrade this block of code to handle the possible exceptions that can occur with `class_getName`:

``` objc
for (unsigned int i = 0; i < classCount; i++) {
    const char* potentialClassName = class_getName(classList[i]);
    if (potentialClassName != NULL &&
        strcmp(potentialClassName, "nil") != 0 &&
        strcmp(potentialClassName, "") != 0) {
        NSString *className = [NSString stringWithUTF8String:potentialClassName];
        if([className hasPrefix:@"TSDK"]) {
            Class class = NSClassFromString(className);
            
            if( [class isEqual:[TSDKCollectionObject class]] == false &&
               [class isEqual:[TSDKShortPropertyTest class]] == false) {
                if(class && [class isSubclassOfClass:[TSDKCollectionObject class]]) {
                    [supportedObjects addObject:class];
                }
            }
        }
    }
}
```

Reply:
Then we can remove line 30/31 in the test for the static list

Reply:
Resolved